Flow.Launcher/HotkeyControl.xaml.cs

```diff
@@ -101,9 +101,9 @@ public async Task SetHotkey(HotkeyModel keyModel, bool triggerValidate = true)
             }
         }
 
-        public void SetHotkey(string keyStr, bool triggerValidate = true)
+        public async Task SetHotkeyAsync(string keyStr, bool triggerValidate = true)
         {
-            SetHotkey(new HotkeyModel(keyStr), triggerValidate);
+            await SetHotkeyAsync(new HotkeyModel(keyStr), triggerValidate);
```

Comment:
I feel like we can simply return `SetHotkeyAsync` but this is also fine

Reply:
We can't return it directly as we want to await the result (since there is a 500ms delay in the other `SetHotkeyAsync`)

Reply:
Oh i mean return the task from the other function.

Reply:
I tried:

```csharp
        public Task SetHotkeyAsync(string keyStr, bool triggerValidate = true)
        {
            return SetHotkeyAsync(new HotkeyModel(keyStr), triggerValidate);
        }
```

but then you can't await this function (since it's not `async`). If we want this to work as a pass through then it looks like:

```csharp
        public async Task<Task> SetHotkeyAsync(string keyStr, bool triggerValidate = true)
        {
            return SetHotkeyAsync(new HotkeyModel(keyStr), triggerValidate);
        }
```

but that's awkward (and would need some kind of double await if you wanted to await this) :)

Reply:
Lol i think you understand async/await a little bit wrong. Any Task can be awaited, regardless whether the method has async signature. It is true that if we remove the async here we are not able to await function here, but since this method is only an overload of the other, we don't actually need to await anything here. 

However, in both case, the one level Task will be able to be awaited.

Reply:
heh. That was my expectation going in, but when I tried the first one it errored out - though I think I had my warnings dialed up too high which caused a warning to show as an error. I've tried again and it works cleanly *shrug*.

Reply:
lol that's common. Just note that what affects whether a function can be awaited is the return type (`Task, Task<T>, ValueTask<T>`), but not the async modifier. The async modifier only allows to use `await` inside the function.